test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/TensorflowTests.cs

```diff
@@ -389,7 +385,7 @@ public void TensorFlowTransformInputOutputTypesTest()
 
             var inputs = new string[] { "f64", "f32", "i64", "i32", "i16", "i8", "u64", "u32", "u16", "u8", "b" };
             var outputs = new string[] { "o_f64", "o_f32", "o_i64", "o_i32", "o_i16", "o_i8", "o_u64", "o_u32", "o_u16", "o_u8", "o_b" };
-            var tfModel = mlContext.Model.LoadTensorFlowModel(modelLocation);
+            using var tfModel = mlContext.Model.LoadTensorFlowModel(modelLocation);
```

Comment:
Consistency/user-expectations:
How will a user know which model types and prediction engines are disposable? Should we convert all models and prediction engines to be disposable?

Seems it should be an all-or-none for consistency.

Reply:
I agree that it should be all or none for consistency. If we go for none, then we need to make the interfaces inherit from IDisposable and we can't do that now without a breaking change. If for "all" then the Dispose() calls for most would become no-ops and only Tensorflow (right now) needs a real Dispose() call. 
This seems like one of those cases that we will need to fix whenever we do a v2 release and can take breaking changes. Until then, I am okay with documenting for the tensorflow related transformers. 

---
In reply to: [396761653](https://github.com/dotnet/machinelearning/pull/4939#discussion_r396761653) [](ancestors = 396761653)